engines/scumm/charset.cpp

```diff
@@ -1678,33 +1616,20 @@ int CharsetRendererMac::getStringWidth(int arg, const byte *text) {
 }
 
 int CharsetRendererMac::getDrawWidthIntern(uint16 chr) const {
-	return _macFonts[_curId]->getCharWidth(chr);
+	return _font->getCharWidth(chr);
 }
 
-// HACK: Usually, we want the approximate width and height in the unscaled
-//       graphics resolution. But for font 1 in Indiana Jones and the Last
-//       crusade we want the actual dimensions for drawing the text boxes.
-
 int CharsetRendererMac::getFontHeight() const {
-	int height = _macFonts[_curId]->getFontHeight();
-
-        // If we ever need the height for font 1 in Last Crusade (we don't at
-	// the moment), we need the actual height.
-	if (_curId == 0 || _vm->_game.id != GID_INDY3)
-		height /= 2;
-
-	return height;
+	return _font->getFontHeight() / 2;
 }
 
 int CharsetRendererMac::getCharWidth(uint16 chr) const {
-	int width = getDrawWidthIntern(chr);
-	return _useRealCharWidth ? width : width / 2;
+	return _font->getCharWidth(chr) / 2;
 }
```

Comment:
This is because we're drawing high-resolution graphics on a low resolution screen. As far as I know, font rendering is close to pixel perfect, though text positioning is something slightly off for centered text in Loom. I think this is as good as it's going to get, at least for now.